### PR TITLE
Save features and load saved features

### LINKED ISSUE #10: Save features
The ability to save extracted features and use previously saved features should be ported into the new OND protocol.

### PR DESCRIPTION
**Pull request by nrsyed**
@waxlamp @cfunk1210 @as6520 

This PR adds functionality for using saved features from the feature extraction step.

Depends on #14. Closes #10.

### CODE REVIEW

File: ond.py [hunk 1]
```diff
@@ -61,59 +65,122 @@ def run_protocol(self, config_):
         )
 
         for test_id in config["test_ids"]:
+            algo_test_params = {
+                "dataset_ids": [],
+                "dataset_root": config["dataset_root"],
+                "image_features": {},
```
Comment (as6520) [inline ond.py]: This should be renamed to `features` since we use it for videos too.

File: ond.py [hunk 2]
```diff
@@ -61,59 +65,122 @@ def run_protocol(self, config_):
...
                     if config["save_features"]:
-                        logging.info(
-                            f"Writing features to {config['save_features']}"
+                        test_features["features_dict"].update(
+                            algo_test_data["features_dict"]
                         )
+                        test_features["logits_dict"].update(
+                            algo_test_data["logits_dict"]
+                        )
+                        if config["feature_extraction_only"]:
```
Comment (as6520) [inline ond.py]: I don't think it is necessary to have `feature_extraction_only` inside the `save_features`.

**Review by as6520 — CHANGES_REQUESTED** [review]

File: ond.py (hunk 1 above)
Comment (nrsyed) [reply ond.py]: Updated in the latest commit.

File: ond.py (hunk 2 above)
Comment (nrsyed) [reply ond.py]: Good catch, I think the check for `feature_extraction_only` should exist at the same indentation level as the check for `save_features`? Updated it to this in the latest commit.

Comment (as6520) [reply ond.py]: yes, looked at the most recent commit and it looks good

**Pull request merged**